src/Compilers/Helpers/ConsoleUtil.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Roslyn.Utilities
+{
+    internal static class ConsoleUtil
+    {
+        private static readonly Encoding s_utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+        /// <summary>
+        /// When <paramref name="utf8Encoding"/> is true then <paramref name="func"/> will be run
+        /// ent while both <see cref="Console.Out"/> and <see cref="Console.Error"/> are set 
+        /// to set to UTF8 encoding.  Otherwise it will be run with the <see cref="Console"/> in
+        /// its current state.
+        /// </summary>
+        internal static T RunWithUtf8Output<T>(bool utf8Encoding, Func<TextWriter, TextWriter, T> func)
```

Comment:
It doesn't feel right to include Utf8 in the name of the function when usage of UTF8 encoding is conditional based on parameter's value. Perhaps name it "RunWithOutput" instead?